packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,568 @@
+      pressedValue: _pressedValue,
+      keys: widget.children.keys.toList(),
+    );
+
+    return Padding(
+      padding: _kHorizontalItemPadding.resolve(Directionality.of(context)),
+      child: UnconstrainedBox(
+        constrainedAxis: Axis.horizontal,
+        child: box,
+      ),
+    );
+  }
+}
+
+class _SegmentedControlRenderWidget<T> extends MultiChildRenderObjectWidget {
+  _SegmentedControlRenderWidget({
+    Key key,
+    List<Widget> children = const <Widget>[],
+    @required this.selectedValue,
+    @required this.pressedValue,
+    @required this.keys,
+  }) : super(
+          key: key,
+          children: children,
+        );
+
+  final T selectedValue;
+  final T pressedValue;
+  final List<T> keys;
+
+  @override
+  RenderObject createRenderObject(BuildContext context) {
+    return new _RenderSegmentedControl<T>(
+      textDirection: Directionality.of(context),
+      selectedValue: selectedValue,
+      pressedValue: pressedValue,
+      keys: keys,
+    );
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, _RenderSegmentedControl<T> renderObject) {
+    renderObject
+      ..textDirection = Directionality.of(context)
+      ..selectedValue = selectedValue
+      ..pressedValue = pressedValue;
+  }
+}
+
+class _SegmentedControlContainerBoxParentData extends ContainerBoxParentData<RenderBox> {}
+
+class _RenderSegmentedControl<T> extends RenderBox
+    with ContainerRenderObjectMixin<RenderBox, ContainerBoxParentData<RenderBox>> {
+  _RenderSegmentedControl({
+    List<RenderBox> children,
+    @required T selectedValue,
+    @required T pressedValue,
+    @required TextDirection textDirection,
+    @required this.keys,
+  })  : _textDirection = textDirection,
+        _selectedValue = selectedValue,
+        _pressedValue = pressedValue {
+    addAll(children);
+  }
+
+  T get selectedValue => _selectedValue;
+  T _selectedValue;
+  set selectedValue(T value) {
+    if (_selectedValue == value) {
+      return;
+    }
+    _selectedValue = value;
+    markNeedsPaint();
+  }
+
+  T get pressedValue => _pressedValue;
+  T _pressedValue;
+  set pressedValue(T value) {
+    if (_pressedValue == value) {
+      return;
+    }
+    _pressedValue = value;
+    markNeedsPaint();
+  }
+
+  TextDirection get textDirection => _textDirection;
+  TextDirection _textDirection;
+  set textDirection(TextDirection value) {
+    if (_textDirection == value) {
+      return;
+    }
+    _textDirection = value;
+    markNeedsLayout();
+  }
+
+  List<T> keys;
+
+  List<RRect> get rects => _rects;
+  final List<RRect> _rects = <RRect>[];
+
+  List<Paint> get fillPaint => _fillPaint;
+  final List<Paint> _fillPaint = <Paint>[];
+
+  static Size _boxSize(RenderBox box) => box == null ? Size.zero : box.size;
+
+  static Rect _boxRect(RenderBox box) {
+    final _SegmentedControlContainerBoxParentData childData = box.parentData;
+    return box == null ? Rect.zero : childData.offset & box.size;
```

Comment:
you check box == null after first dereferencing box. if it's ever null, this would crash, not return zero.

Reply:
having said that, you never call this with a non-null value, so checking for null is not useful.